pkg/test/expectations/expectations.go

```diff
@@ -190,3 +190,9 @@ func ExpectReconcileSucceeded(ctx context.Context, reconciler reconcile.Reconcil
 	_, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: key})
 	Expect(err).ToNot(HaveOccurred())
 }
+
+func ExpectReconcileSucceededWithResult(ctx context.Context, reconciler reconcile.Reconciler, key client.ObjectKey, result reconcile.Result) {
```

Comment:
I'd rather keep this logic inline until we use it broadly. The concern is about bloating the expectations library to have so many methods that it results in tests written many ways using similar helpers for the same effect.

Reply:
Given the back and forth and the time it took to get this RequeueAfter logic right, I fully expect (pun intended) us to verify this in other controllers too which is why I added it here. 

Don't feel too strongly about it though

Reply:
What about returning a result that can be optionally used for future assertions? Some other expectations work this way:
```
func ExpectReconcileSucceeded(ctx context.Context, reconciler reconcile.Reconciler, key client.ObjectKey) reconcile.Result {
	result, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: key})
	Expect(err).ToNot(HaveOccurred())
        return result
}
```
This won't force any existing code to change, either.